pytorch_lightning/strategies/deepspeed.py

```diff
@@ -357,6 +357,8 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
         self.setup_optimizers(trainer)
         self.setup_precision_plugin()
+        if self.root_device == torch.device("cpu"):
+            raise MisconfigurationException(f"Unsupported accelerator for DeepSpeed: {self.root_device}")
```

Comment:
we should probably fail before spawning any processes. That means checking
`isinstance(self.accelerator, CPUAccelerator)` in the init. That's my proposal. 
Adding a simple test in `tests/strategies/test_deepspeed_strategy.py` would be great!

Reply:
That's a good point, @awaelchli - I agree! @myxik - Please reach out if you have any queries about this. Let's make sure that it's tested as well :) (as Adrian pointed out)

Reply:
Thanks guys I will work on it!

Reply:
It seems that I cannot import `accelerators` in `strategy` folders due to circular imports with `pytorch_lightning.utilities`. I would be thankful if there is any suggestion to substitute usage of `GPUAccelerator`

Reply:
You could move the import locally to where you need it 😃

Reply:
Is there a reason why we haven't done this?

Reply:
we can't check inside init for the cases where the strategy is passed as `Deepspeed` instance.

Reply:
Ok. Once we have lazy init https://github.com/PyTorchLightning/pytorch-lightning/issues/7650 we should do it this way however.